evap/staff/tests/test_forms.py

```diff
@@ -554,6 +557,39 @@ def test_staff_can_select_proxy_user(self):
         form = CourseForm(instance=course)
         self.assertIn(proxy_user, form.fields["responsibles"].queryset)
 
+    def test_prevent_contribution_deletion_with_answers(self):
+        """
+        When answers for a contribution already exist, it should not be possible to remove that contribution.
+        """
+        self.assertEqual(
+            set(Answer.__subclasses__()),
+            {RatingAnswerCounter, TextAnswer},
+            "This requires an update if a new answer type is added",
+        )
+        evaluation = baker.make(Evaluation)
+        user1 = baker.make(UserProfile)
+        questionnaire = baker.make(Questionnaire, type=Questionnaire.Type.CONTRIBUTOR)
+        contribution1 = baker.make(
```

Comment:
You could inline the creation of `contributor` and omit the mention of the questionnaire altogether. Secondly, there isn't a `contribution2`, so why is this one called `contribution1`?

Reply:
(for the contributor, ` _fill_optional=['contributor']` would completely remove `user1`)

Reply:
Yep makes sense :)